src/adapters/discord.js

```diff
@@ -1,142 +1,112 @@
 const Discord = require('discord.js');
 const debug = require('debug')('bot:adapter:discord');
 
-const isString = require('lodash/isString');
-const isObject = require('lodash/isObject');
-const isArray = require('lodash/isArray');
-const isEmpty = require('lodash/isEmpty');
+const merge = require('lodash/merge');
+const pick = require('lodash/pick');
 
 // const { PREFIX } = require('./../config');
 const { discord: discordCfg } = require('../config');
 
 const discordAdapter = () => {};
 
-discordAdapter.__INIT__ = function ({ process }) {
+discordAdapter.__INIT__ = function (ctx) {
     const discordBot = new Discord.Client();
 
     discordBot.on('ready', () => {
         debug('Discrodjs ready');
     });
 
-    discordBot.on('message', (msg) => {
+    const handler = async (output) => {
+        const { message, to, reactions } = output;
+        const [, channelId, msgId] = to;
+        let { embed } = output;
+
+        const channel = discordBot.channels.find(ch => ch.id === channelId);
+
+        if (msgId && reactions) {
+            const msg = await channel.fetchMessage(msgId);
+
+            await reactions.reduce(
+                (prev, reaction) => prev.then(() => msg.react(reaction).catch((e) => {
+                    console.error(e.message);
+                })),
+                Promise.resolve(),
+            );
+        }
+
+        if (embed) {
+            const {
+                title,
+                description,
+                url,
+                fields = [],
+            } = embed;
+
+            embed = new Discord.RichEmbed();
+            embed.setColor(discordCfg.color);
```

Comment:
may be we wanna differ it, depending on what kind of message we're sending? I mean: one for missions, one for polls, etc, just for visual clarification

Reply:
In future, yep! and add option to embed